### DISCUSSION TITLE
Emergency power status

### DISCUSSION THREAD

**Discussion by just-grizzle** [torbennehmer/hacs-e3dc#182] [Ideas]
Hello all,

my E3DC system has been installed recently and I'm starting to integrate it to HA. As the emergency power function of this system, I want to come up with the Feature request described [in this issue](https://github.com/torbennehmer/hacs-e3dc/issues/96).

Knowing the limitations, I configured the HA Integration using the `local.user`. As far as I can tell, this setup works perfectly fine, even when internet access is blocked.

Next step will be to add a UPS to my network gear and home server. For my understanding, I would then be able to monitor the E3DC even on power outages. So knowing when the emergency power is active would make sense.

Is there anything else to consider? Or is the feature not planned because of the lack of testing units?

**Comment by torbennehmer** [comment]
Does your local.user access work when you're online? To my understanding, this should not be the case (which is why I never actually tried it, to my shame). If you can authenticate working around the E3DC portal authentication, I'd be really interested in this.

As for your question:

> Next step will be to add a UPS to my network gear and home server. For my understanding, I would then be able to monitor the E3DC even on power outages. So knowing when the emergency power is active would make sense.

Yes, you will need an UPS. E3DC needs 1-2 seconds to actually switch a big relay from A to B, so that your internal power grid is severed from the external one. Without the UPS, your servers will crash because of that.

If you can bridge that gap, your approach would perfectly fine - if we can get the authentication across.

So I'd like your input on this regarding the local.user authentication while online. :) If that works, setting things up would be *a lot* easier. I don't have time to test this in the coming days, so if you have feedback, please comment this here, with as much info as possible. Then I'll see what we can make of it.

**Reply by just-grizzle** [reply]
My Home Assistant instance has been connected with the local account since my last post 3 weeks ago.

For my test I blocked internet access for the E3DC and could then connect via local.user. So maybe the login worked because of the offline state and since then there were only token refreshes. 
Not sure what happens with the session when HA restarts. I restarted it several times.

I'm trying to figure out how to properly test the login procedure without the my current session affecting the results.
